src/tests/ftest/util/pool_security_test_base.py

```diff
@@ -0,0 +1,438 @@
+        port = self.params.get("port", "/run/server_config/*", 10001)
+        get_acl_file = self.params.get("acl_file", "/run/pool_acl/*",
+                                       "acl_test.txt")
+        acl_file = os.path.join(self.tmp, get_acl_file)
+        num_user = self.params.get("num_user", "/run/pool_acl/*")
+        num_group = self.params.get("num_group", "/run/pool_acl/*")
+        servers_with_ports = [
+            "{}:{}".format(host, port) for host in self.hostlist_servers]
+        dmg.hostlist.update(",".join(servers_with_ports), "dmg.hostlist")
+        self.log.info("  (1)dmg= %s", dmg)
+
+        # (2)Generate acl file with permissions
+        self.log.info("  (2)Generate acl file with user/group permissions")
+        permission_list = self.create_pool_acl(num_user,
+                                               num_group,
+                                               current_user_acl,
+                                               acl_file)
+
+        # (3)Create a pool with acl
+        self.log.info("  (3)Create a pool with acl")
+        dmg.action_command.acl_file.value = acl_file
+        dmg.exit_status_exception = False
+        result = dmg.run()
+
+        # (4)Verify the pool create status
+        self.log.info("  (4)dmg.run() result=\n%s", result)
+        if "ERR" not in result.stderr:
+            uuid, svc = dmg_utils.get_pool_uuid_service_replicas_from_stdout\
+                (result.stdout)
+        else:
+            self.fail("##(4)Unable to parse pool uuid and svc.")
+
+        # (5)Get the pool's acl list
+        #    dmg pool get-acl --pool <UUID>
+        self.log.info("  (5)Get a pool's acl list by: "
+                      "dmg pool get-acl --pool --hostlist")
+        pool_acl_list = self.get_pool_acl_list(uuid)
+        self.log.info(
+            "   pool original permission_list: %s", permission_list)
+        self.log.info(
+            "   pool get_acl  permission_list: %s", pool_acl_list)
+
+        # (6)Verify pool read operation
+        #    daos pool query --pool <uuid>
+        self.log.info("  (6)Verify pool read by: daos pool query --pool")
+        self.verify_pool_readwrite(svc, uuid, "read", expect=read)
+
+        # (7)Verify pool write operation
+        #    daos continer create --pool <uuid>
+        self.log.info("  (7)Verify pool write by: daos continer create --pool")
+        self.verify_pool_readwrite(svc, uuid, "write", expect=write)
+        if secondary_grp:
+            self.log.info("  (7-0)Verifying verify_pool_acl_prim_sec_groups")
+            self.verify_pool_acl_prim_sec_groups(pool_acl_list, acl_file,\
+                uuid, svc)
+
+        # (8)Cleanup user and destroy pool
+        self.log.info("  (8)Cleanup user and destroy pool")
+        self.cleanup_user_group(num_user, num_group)
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.request.value = "pool"
+        dmg.action.value = "destroy --pool={}".format(uuid)
```

Comment:
Ideally, we should use
`dmg.action.value = "destroy"`
`dmg.get_action_command()`
`dmg.action_command.pool.value = uuid`